mssql-plugin/src/main/java/io/cdap/plugin/mssql/SqlFieldsValidator.java

```diff
@@ -35,17 +35,29 @@ public boolean isFieldCompatible(Schema.Field field, ResultSetMetaData metadata,
     Schema.LogicalType fieldLogicalType = fieldSchema.getLogicalType();
 
     int sqlType = metadata.getColumnType(index);
+    String sqlTypeName = metadata.getColumnTypeName(index);
     boolean isSigned = metadata.isSigned(index);
     int precision = metadata.getPrecision(index);
 
     // Handles datetime datatypes
     // Case when Timestamp maps to datetime
     // Case when Datetime2 maps to datetime
-    // Case when DatetimeOffset maps to datetime
-    if ((sqlType == Types.TIMESTAMP || sqlType == SqlServerSourceSchemaReader.DATETIME_OFFSET_TYPE)
-            && fieldLogicalType.equals(Schema.LogicalType.DATETIME)) {
+    if (sqlType == Types.TIMESTAMP && fieldLogicalType.equals(Schema.LogicalType.DATETIME)) {
```

Comment:
This change is not backward compatible. We should not break the compatibility of DateTimeOffset with DateTime type at the sink as it might regress any existing use case.

Reply:
Added the fix to ensure backward compatibility